### DISCUSSION TITLE
How can I solve reCapcha manually and finish rest of my code.

### DISCUSSION THREAD

**Discussion by SergeyGuschin** [seleniumbase/SeleniumBase#1826] [General]
Hello Michael. Thank you again for your excellent work and help. I need your help again. For a couple of days, I tried to solve the reCaptcha problem. My algorithm is to:
1 Enter to reCaptcha iFrame click the reCaptcha checkbox
2 check if a new reCaptcha challenge window with pictures opened entered to iFrame and wait to reCaptcha challenge window is gone
3 Manually solved reCaptcha
4 Check if the reCaptcha challenge window disappeared exit from iFrame and proceed to the next step in code.
I can do all three steps but when I try to wait_for_element_notvisible I have an Exception. Can you please help me to solve my puzzle? Thank you.

**Comment by mdmintz** [comment]
What exception do you have when calling ``wait_for_element_not_visible()``? That method will fail if the element is still visible.

The best way to handle captchas is to try avoiding them. You might be able to use ``--uc`` mode to avoid them, which can make the browser undetectable to services that try to detect if a web browser is being controlled by selenium. Here's one of the examples that uses that: https://github.com/seleniumbase/SeleniumBase/blob/master/examples/uc_cdp_events.py

**Comment by SergeyGuschin** [comment]
I know that the web store I am testing is not using a method to detect if the browser using selenium webdriver or not. But in order to test the registration of a new user, I need to solve the reCaptcha. After I wrote the question, I solved my problem )). My problem was I tried to enter an iframe from the first reCaptcha iframe and wait to reChapcha puzzle window with pictures not visible anymore. But it doesn't work this way because the element stays in DOM and is always visible even after the ReCaptcha puzzle is solved.  All I need to solve my problem is to click on the checkbox in the first iframe and wait for a  green checkmark to appear in this iframe, This checkmark appears there only after the ReCaptcha puzzle with pictures is solved. So I wait for this element and everything works like a charm. The only thing  I need to increase the waiting time for the element to be present. Because it is not enough time to solve pictures puzzle manually. But as soon as the reCaptha picture puzzle is solved, the code proceeds to the next step. Thank you very much for your quick replies and your help. She helps me a lot.
